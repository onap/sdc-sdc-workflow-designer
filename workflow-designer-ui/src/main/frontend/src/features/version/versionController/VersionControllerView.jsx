/*
* Copyright Â© 2018 European Support Limited
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
 *      http://www.apache.org/licenses/LICENSE-2.0
*
 * Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import React, { Component } from 'react';
import PropTypes from 'prop-types';

import ActionButtons from 'features/version/versionController/views/ActionButtons';
import VersionContainer from 'features/version/versionController/views/VersionsContainer';
import WorkflowTitle from 'features/version/versionController/views/WorkflowTitle';
import { versionState as versionStateType } from 'features/version/versionConstants';

export default class VersionControllerView extends Component {
    static propTypes = {
        location: PropTypes.object,
        workflowName: PropTypes.string,
        currentWorkflowVersion: PropTypes.object,
        viewableVersions: PropTypes.arrayOf(Object),
        getVersions: PropTypes.func,
        versionsList: PropTypes.array,
        history: PropTypes.object,
        getOverview: PropTypes.func,
        match: PropTypes.object,
        savedParams: PropTypes.object,
        saveParamsToServer: PropTypes.func,
        workflowId: PropTypes.string,
        versionState: PropTypes.string,
        certifyVersion: PropTypes.func,
        changeVersion: PropTypes.func,
        getIOErrors: PropTypes.bool
    };

    constructor(props) {
        super(props);
    }

    routeToOverview = () => {
        const { history, match } = this.props;
        const workflowId = match.params.workflowId;
        history.push('/workflow/' + workflowId + '/overview');
    };

    sendSaveParamsToServer = () => {
        const { savedParams, saveParamsToServer, workflowId } = this.props;
        saveParamsToServer({ params: savedParams, workflowId });
    };

    certifyVersion = () => {
        const {
            certifyVersion,
            workflowId,
            currentWorkflowVersion
        } = this.props;
        certifyVersion({ workflowId, versionId: currentWorkflowVersion.id });
    };

    versionChangeCallback = versionId => {
        const { changeVersion, workflowId } = this.props;
        changeVersion({ versionId, workflowId });
    };

    undoClickCallback = () => {
        const {
            currentWorkflowVersion,
            changeVersion,
            workflowId
        } = this.props;
        changeVersion({ versionId: currentWorkflowVersion.id, workflowId });
    };

    render() {
        const {
            currentWorkflowVersion,
            workflowName,
            versionsList,
            versionState,
            getIOErrors
        } = this.props;
        let isCertifyDisable =
            versionState &&
            versionState.toLowerCase() === versionStateType.CERTIFIED;
        return (
            <div className="version-controller-bar">
                <WorkflowTitle workflowName={workflowName} />
                <div className="vc-container">
                    <VersionContainer
                        currentWorkflowVersion={currentWorkflowVersion}
                        viewableVersions={versionsList}
                        onOverviewClick={this.routeToOverview}
                        onVersionSelectChange={this.versionChangeCallback}
                    />
                    <ActionButtons
                        saveDisabled={isCertifyDisable || getIOErrors}
                        onSaveClick={this.sendSaveParamsToServer}
                        certifyDisabled={isCertifyDisable}
                        onCertifyClick={this.certifyVersion}
                        onUndoClick={this.undoClickCallback}
                    />
                </div>
            </div>
        );
    }
}

VersionControllerView.defaultProps = {
    getVersions: () => {}
};
