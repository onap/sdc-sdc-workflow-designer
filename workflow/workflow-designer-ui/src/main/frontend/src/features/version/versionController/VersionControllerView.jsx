/*
* Copyright Â© 2018 European Support Limited
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
 *      http://www.apache.org/licenses/LICENSE-2.0
*
 * Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import React, { Component } from 'react';
import PropTypes from 'prop-types';

import ActionButtons from './views/ActionButtons';
import VersionContainer from './views/VersionsContainer';
import WorkflowTitle from './views/WorkflowTitle';

export default class VersionControllerView extends Component {
    static propTypes = {
        location: PropTypes.object,
        workflowName: PropTypes.string,
        currentWorkflowVersion: PropTypes.string,
        viewableVersions: PropTypes.arrayOf(Object),
        callForAction: PropTypes.func,
        getVersions: PropTypes.func,
        versionsList: PropTypes.array,
        history: PropTypes.object,
        getOverview: PropTypes.func,
        match: PropTypes.object
    };

    constructor(props) {
        super(props);
    }

    dynamicDispatcher = (action, payload) => {
        const { history, callForAction } = this.props;
        const actionName =
            typeof action === 'object'
                ? action.target.attributes.actionType.value
                : action;
        let pageName = history.location.pathname.split('/').pop();
        callForAction(pageName + '/' + actionName, payload);
    };

    routeToOverview = () => {
        const { history, match } = this.props;
        const workflowId = match.params.workflowId;
        history.push('/workflow/' + workflowId + '/overview');
    };

    render() {
        const {
            currentWorkflowVersion,
            workflowName,
            versionsList
        } = this.props;
        return (
            <div className="version-controller-bar">
                <WorkflowTitle workflowName={workflowName} />
                <div className="vc-container">
                    <VersionContainer
                        dynamicDispatcher={this.dynamicDispatcher}
                        currentWorkflowVersion={currentWorkflowVersion}
                        viewableVersions={versionsList}
                        onOverviewClick={this.routeToOverview}
                    />
                    <ActionButtons dynamicDispatcher={this.dynamicDispatcher} />
                </div>
            </div>
        );
    }
}

VersionControllerView.defaultProps = {
    getVersions: () => {},
    callForAction: () => {}
};
