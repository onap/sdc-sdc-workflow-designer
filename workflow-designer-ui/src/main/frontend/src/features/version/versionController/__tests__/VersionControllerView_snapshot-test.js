/*
* Copyright Â© 2018 European Support Limited
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
 *      http://www.apache.org/licenses/LICENSE-2.0
*
 * Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import React from 'react';
import renderer from 'react-test-renderer';

import VersionControllerView from 'features/version/versionController/VersionControllerView';

describe('Version Controller View Snapshot', () => {
    it('renders correctly', () => {
        const versionList = [
            {
                id: '7b5f6b086613470985082df2c0f6c713',
                name: '1.0',
                description:
                    'Initial version, bug fix for previous version that fixed an exception when the port was occupied',
                status: 'Draft',
                creationTime: 1530687330460,
                modificationTime: 1530687330575,
                archivedStatus: 'ACTIVE'
            },
            {
                id: '7b5f6b086613470985082df2c0f6c666',
                name: '2.0',
                description:
                    'Test version, bug fix for previous version that fixed an exception when the port was occupied',
                status: 'Draft',
                creationTime: 1530687330461,
                modificationTime: 1530687330576,
                archivedStatus: 'ACTIVE',
                baseId: '7b5f6b086613470985082df2c0f6c713'
            }
        ];
        const tree = renderer
            .create(
                <VersionControllerView
                    viewableVersions={versionList}
                    currentWorkflowVersion={versionList[0]}
                />
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
