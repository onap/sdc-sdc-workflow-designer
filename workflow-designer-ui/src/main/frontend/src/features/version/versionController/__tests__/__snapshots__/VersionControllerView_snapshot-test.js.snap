// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Version Controller View Snapshot renders correctly 1`] = `
<div
  className="version-section-wrapper"
>
  <div
    className="version-status-container"
  >
    version
    <select
      className="version-selector"
      data-test-id="vc-versions-select-box"
      onChange={[Function]}
      value="7b5f6b086613470985082df2c0f6c713"
    >
      <option
        data-test-id="vc-version-option"
        value="7b5f6b086613470985082df2c0f6c713"
      >
        1.0 DRAFT
      </option>
      <option
        data-test-id="vc-version-option"
        value="7b5f6b086613470985082df2c0f6c666"
      >
        2.0 Draft
      </option>
    </select>
    <span
      className="version-selector-more-versions"
      data-test-id="vc-versions-page-link"
      onClick={undefined}
    >
      viewOverview
    </span>
  </div>
</div>
`;
